src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -323,7 +315,34 @@ def should_use_specific_entry(matrix_combo, specific_entry_matrix):
     )
 
 
-def make_dependency_files(parsed_config, config_file_path, to_stdout):
+def name_with_cuda_suffix(name, cuda_version=None, cuda_suffix="-cu"):
```

Comment:
We can (and do) put the `-cuXX` names in the `dependencies.yaml` matrix entries. This is explicitly about the `name` key of the `pyproject.toml`.

Overall `pyproject.toml` should probably be fully auto-generated instead of read and mutated.

Reply:
@vyasr where do you think the logic to define the `project.name` key in the `pyproject.toml` should live?

Reply:
When you say fully auto-generated, what are you envisioning? I could see using something like a Jinja templated pyproject.toml.in and filling in a suitable set of fields, perhaps. There are large swathes of the file that are necessary for things like 1) running linters, 2) specifying "extra" dependency lists, 3) configuring build backends, and more that have to be encoded somewhere else. None of those should be dependency-file-generator's responsibility, and at least the first one requires the file to already exist somewhere in the repo for normal usage.

Reply:
Regarding the `project.name`, I honestly do not know. My approach to taking the magic out of the wheel building process so far has been very incremental, fixing one problem at a time. I've been viewing the name as sort of a final frontier, one for which I don't have a good answer for yet unfortunately. I don't think a tool dedicated to dependency management is the right place to put that, though.